From 26cbac44b5f8e957d936ecc735ac108aa37b9588 Mon Sep 17 00:00:00 2001
From: "Chen, Gang G" <gang.g.chen@intel.com>
Date: Mon, 11 Dec 2023 21:40:13 +0800
Subject: [PATCH 36/54] Increase TPM pause value

It will reduce the TPM FlushContext failure rate

Tracked-On: OAM-114168
Signed-off-by: Chen, Gang G <gang.g.chen@intel.com>
---
 libedk2_tpm/Tpm2Ptp.c | 4 ++--
 libedk2_tpm/Tpm2Tis.c | 8 ++++----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/libedk2_tpm/Tpm2Ptp.c b/libedk2_tpm/Tpm2Ptp.c
index d88be47..65e9ea8 100644
--- a/libedk2_tpm/Tpm2Ptp.c
+++ b/libedk2_tpm/Tpm2Ptp.c
@@ -86,12 +86,12 @@ PtpCrbWaitRegisterBits (
   UINT32                            RegRead;
   UINT32                            WaitTime;
 
-  for (WaitTime = 0; WaitTime < TimeOut; WaitTime += 30) {
+  for (WaitTime = 0; WaitTime < TimeOut; WaitTime += 100) {
     RegRead = MmioRead32 ((UINTN)Register);
     if ((RegRead & BitSet) == BitSet && (RegRead & BitClear) == 0) {
       return EFI_SUCCESS;
     }
-    pause_us(30);
+    pause_us(100);
   }
   return EFI_TIMEOUT;
 }
diff --git a/libedk2_tpm/Tpm2Tis.c b/libedk2_tpm/Tpm2Tis.c
index 5c2f0f8..beec46d 100644
--- a/libedk2_tpm/Tpm2Tis.c
+++ b/libedk2_tpm/Tpm2Tis.c
@@ -64,12 +64,12 @@ TisPcWaitRegisterBits (
   UINT8                             RegRead;
   UINT32                            WaitTime;
 
-  for (WaitTime = 0; WaitTime < TimeOut; WaitTime += 30) {
+  for (WaitTime = 0; WaitTime < TimeOut; WaitTime += 100) {
     RegRead = MmioRead8 ((UINTN)Register);
     if ((RegRead & BitSet) == BitSet && (RegRead & BitClear) == 0) {
       return EFI_SUCCESS;
     }
-    pause_us(30);
+    pause_us(100);
   }
   return EFI_TIMEOUT;
 }
@@ -111,8 +111,8 @@ TisPcReadBurstCount (
     if (*BurstCount != 0) {
       return EFI_SUCCESS;
     }
-    pause_us(30);
-    WaitTime += 30;
+    pause_us(100);
+    WaitTime += 100;
   } while (WaitTime < TIS_TIMEOUT_D);
 
   return EFI_TIMEOUT;
-- 
2.25.1

